libkineto/src/output_json.cpp

```diff
@@ -244,6 +244,43 @@ void ChromeTraceLogger::handleGenericActivity(
       op.traceSpan()->name, op.traceSpan()->iteration, separator,
       op_metadata);
   // clang-format on
+
+  if (op.unique_op_id != 0) {
```

Comment:
Change to check for linkedActivity() != nullptr after making changes suggested above, and add handleGenericLink() instead taking two activities (start and end).

Reply:
Thanks. Done.